packages/repository/src/decorators/relation.ts

```diff
@@ -34,10 +34,11 @@ export class RelationMetadata {
  * @returns {(target:any, key:string)}
  */
 export function relation(definition?: Object) {
-  return function(target: any, key: string) {
-    // Apply model definition to the model class
-    Reflector.defineMetadata(RELATION_KEY, definition, target, key);
-  };
+  // Apply relation definition to the model class
+  return <PropertyDecorator>PropertyDecoratorFactory.getDecorator(
```

Comment:
Why do we have to cast the result of `PropertyDecoratorFactory.getDecorator` to `PropertyDecorator`? Since we are calling `PropertyDecoratorFactory`, there should have been already enough type information for `tsc` to let it know that `getDecorator` is returning a `PropertyDecorator`.

I think this points out a flaw in our factory design.